### PR TITLE
Reverb settings adjustements

### PR DESCRIPTION
**Pull request by jjceresa**
The PR corresponding to issue [Reverb roomsize issue and level/width dependency #279](https://github.com/FluidSynth/fluidsynth/issues/279)

### CODE REVIEW

File: src/rvoice/fluid_rev.c [hunk 1]
```diff
@@ -481,11 +481,31 @@ fluid_revmodel_processmix(fluid_revmodel_t* rev, fluid_real_t *in,
...
-
+  
+  /* The stereo amplitude equation  (wet1 and wet1 below) have a 
+  tendency to produce high amplitude with high width value ( 1 < width < 100).
+  This result in a unwanted output clipped by the audio card.
+  To avoid this dependency, we divide by (1 + rev->width * scalelevelwidth)
+  Actually, with a scalelevelwidthof 0.2, (regardless of level setting), 
+  the output amplitude (wet) seems rather independent of width setting */	
+  rev->wet = (rev->level * scalewet) / 
+             (1.0f + rev->width * scalelevelwidth);
```
Comment (derselbst) [inline src/rvoice/fluid_rev.c]: Looks like `rev->wet` is only use in the local scope here. It should be removed from `struct _fluid_revmodel_t` then.

File: src/rvoice/fluid_rev.c [hunk 2]
```diff
@@ -481,11 +481,31 @@ fluid_revmodel_processmix(fluid_revmodel_t* rev, fluid_real_t *in,
...
 
 static void
+/* scalelevelwidth is compensation weigth factor to get output
+   amplitude rather  independent of width setting.
+   0: the output amplitude is fully dependant of width setting.
+   >0: the output amplitude is less dependant of width setting.
+   With a scalelevelwidthof of 0.2 the output amplitude is 
+   rather independent of width setting.
+*/
+#define scalelevelwidth 0.2f
```
Comment (derselbst) [inline src/rvoice/fluid_rev.c]: This is a pretty unsuitable location to place a define. Please place it some lines above (around 242) where the other `scale*` defines are located.

File: src/rvoice/fluid_rev.c (hunk 1 above)
Comment (jjceresa) [reply src/rvoice/fluid_rev.c]: Yes. Since the arrival of rev->level,   rev->wet need to be removed.

**Comment by derselbst** [comment]
I had a listen. I actually like it, it avoids that the reverb gets too noisy. However now it seems that when setting `width==10` it sounds pretty much the same like when setting `width==100` (while `level==1`). Not sure if a user actually relies on the effect of this range.

**Comment by jjceresa** [comment]
> it avoids that the reverb gets too noisy

Yes, without the compensation factor (i.e `scale_wet_width = 0`), the `output amplitude` dependency of `Width` value was making the reverb very noisy (particularly for high value of Width).

> However now it seems that when setting width==10 it sounds pretty much the same like when setting width==100 (while level==1).

Yes , in general regardless of the compensation factor presence  , the perceived effect of whole Width range [0..100] could be roughly cut  into 2 ranges (ie `wRange1 [0..10]` and` wRange2 [10..100]`). `wRange1` values brings a more prominent perceived  `Width effect` than `wRange2` values (probably due to the logarithmic response of our ear). This remark is more obvious with the presence of compensation (i.e  `scale_wet_width > 0`). 

> Not sure if a user actually relies on the effect of this range.

Not sure too. The upper limit of the range feels excessive.

**Comment by derselbst** [comment]
Ok, I'll merge this for master only.

**Comment by jjceresa** [comment]
Many thanks for fixing grammar.

**Pull request merged**